### PR TITLE
ROX-15925: adds deprecation notices for KernelModule collection

### PR DESCRIPTION
**Pull request by Stringy**
## Description

Adds deprecation notices for kernelmodule collection to:
- operator configuration
- helm chart documentation
- CHANGELOG

## Checklist
~- [ ] Investigated and inspected CI test results~
~- [ ] Unit test and regression tests added~
- [x] Evaluated and added CHANGELOG entry if required
~- [ ] Determined and documented upgrade steps~
~- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))~

User facing docs PR: https://github.com/openshift/openshift-docs/pull/57406 

If any of these don't apply, please comment below.

## Testing Performed

Documentation changes only

### CODE REVIEW

**Review by janisz — COMMENTED** [review]
Looks like more changes are required to pass `check generated`

**Comment by Stringy** [comment]
Given that this is purely documentation changes, I think it's safe to ignore the `gke-postgres-upgrade-tests` failure

**Review by janisz — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -17,7 +17,7 @@ Please avoid adding duplicate information across this changelog and JIRA/doc inp
 
 - ROX-14336: product `BuildDate` attribute was removed. It won't be returned by
 `/debug/versions.json` endpoint and `roxctl version --json` command.
-- ROX-12750: As announced in 3.73.0 (ROX-11101), some permissions for permission sets are being grouped for simplification. The deprecation process will remove and replace the deprecated permissions with the replacing permission as listed below. The access level granted to the replacing permission will be the lowest among all access levels of the replaced permissions. 
+- ROX-12750: As announced in 3.73.0 (ROX-11101), some permissions for permission sets are being grouped for simplification. The deprecation process will remove and replace the deprecated permissions with the replacing permission as listed below. The access level granted to the replacing permission will be the lowest among all access levels of the replaced permissions.
```
Comment (janisz) [inline CHANGELOG.md:20]: I found following sentences hard to understand. 
- `The deprecation process will remove and replace the deprecated permissions with the replacing permission as listed below.`
- `The access level granted to the replacing permission will be the lowest among all access levels of the replaced permissions.`

Why:
- `The deprecation process` this is new entity that is not described anywhere or at least I don't know where :) 
- sentences are quite long 
- many repetitions of `permission`

How about something like this:
> This means that certain permissions will be removed and replaced with new ones as listed below. The new permission will have the lowest access level among all the permissions being replaced.

**Pull request merged**